lib/SILOptimizer/Mandatory/TFUtilities.cpp

```diff
@@ -221,27 +222,19 @@ SILType tf::convertElementTypeToTensorValueType(SILType ty) {
   if (isTensorFlowValue(ty))
     return ty;
 
-  return convertElementTypeToTensorValueType(ty.getASTType(), ty.getASTContext());
+  return convertElementTypeToTensorValueType(ty.getASTType(),
+                                             ty.getASTContext());
 }
 
 /// Looks up a function in the current module. If it exists, returns it.
 /// Otherwise, attempt to link it from imported modules. Returns null if such
 /// function name does not exist.
 SILFunction *tf::lookupOrLinkFunction(StringRef name, SILModule &module) {
-  auto *fn = module.lookUpFunction(name);
-  if (!fn)
-    fn = module.findFunction(name, SILLinkage::PublicExternal);
-  assert(fn);
-  if (fn->isExternalDeclaration()) {
-    bool loaded = module.loadFunction(fn);
-    assert(loaded);
-    (void)loaded;
-    // linkFunction() can return false if this function has already been
-    // deserialized and its body available in `module`.
-    module.linkFunction(fn);
-  }
-  assert(!fn->isExternalDeclaration());
-  assert(fn->isDefinition());
+  assert(!name.empty());
+  if (auto *localFn = module.lookUpFunction(name))
+    return localFn;
+  auto *fn = module.findFunction(name, SILLinkage::PublicExternal);
+  module.loadFunction(fn);
```

Comment:
It's only used within TFPartition on the TF branch, but it could be useful elsewhere. Would you suggest that I check it in to `SILModule` or somewhere else?

Reply:
Well, I would maybe look at existing usages of lookupFunction/findFunction/loadFunction, and if they fit the same or similar pattern, use the new utility method there. However, if this is the only real usage, I wouldn't bother moving it into SILModule. However this is entirely optional, feel free to leave it as-is.

Reply:
Ok! I looked at a couple of call sites of `lookUpFunction` and `findFunction`, and they don't seem to match the logic of this function. I'll revisit this later when we upstream things to master.